api/v1beta1/nodeconfig_types.go

```diff
@@ -78,3 +80,39 @@ type NodeConfigList struct {
 func init() {
 	SchemeBuilder.Register(&NodeConfig{}, &NodeConfigList{})
 }
+
+// ConvertTo converts this v1beta1 to v1beta2. (upgrade)
+func (src *NodeConfig) ConvertTo(dstRaw conversion.Hub) error {
+
+	dst := dstRaw.(*v1beta2.NodeConfig)
+	dst.ObjectMeta = src.ObjectMeta
+
+	dst.Spec.KernelParameters = src.Spec.KernelParameters
+	dst.Spec.KernelModules = src.Spec.KernelModules
+	dst.Spec.SystemdUnits = src.Spec.SystemdUnits
+	dst.Spec.SystemdOverrides = src.Spec.SystemdOverrides
+	dst.Spec.Hosts = src.Spec.Hosts
+	dst.Spec.AptPackages = src.Spec.AptPackages
+	dst.Spec.BlockInFiles = src.Spec.BlockInFiles
+	dst.Spec.NodeSelector = src.Spec.NodeSelector
+
+	return nil
+}
+
+// ConvertFrom converts from the Hub version (v1beta1) to (v1alpha1). (downgrade)
+func (dst *NodeConfig) ConvertFrom(srcRaw conversion.Hub) error {
```

Comment:
It is needed for the webhook to work correctly

Reply:
but why is v1beta2 the hub version? shouldn't it be the lowest version?

Reply:
The Hub version is the one that all others (spokes) will be converted to

https://book.kubebuilder.io/multiversion-tutorial/conversion-concepts.html

Reply:
Since v1beta2 adds a field, if v1beta1 was the hub version we would lose that field when converting to v1beta1.

We basically have to choose between adding that field to v1beta1 or having v1beta2 as a hub version. As I understood it we could not modify v1beta1 so I chose this.

We would eventually move the hub version to v1beta3 if it added more fields not present in other versions

https://book.kubebuilder.io/multiversion-tutorial/api-changes